### PR TITLE
Added a log message when sharing a FlutterEngine across multiple FlutterViewControllers.

### PR DESCRIPTION
**Pull request by gaaclarke**
issue: https://github.com/flutter/flutter/issues/52455

### CODE REVIEW

**Comment by xster** [comment]
Add to https://cs.opensource.google/flutter/engine/+/master:shell/platform/darwin/ios/framework/Headers/FlutterViewController.h;l=71&ss=flutter as well. e.g. if you give me `FlutterEngine` that's already used, we'll crash (we don't have to do it in the implementation right now but we should at some point).

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -144,6 +144,11 @@ - (instancetype)initWithEngine:(FlutterEngine*)engine
   self = [super initWithNibName:nibName bundle:nibBundle];
   if (self) {
     _viewOpaque = YES;
+    if (engine.viewController) {
+      FML_LOG(ERROR) << "Attempting to share a FlutterEngine across multiple "
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: The supplied FlutterEngine << engine << is already used with FlutterViewController instance << engine.ViewController <<. One instance of the FlutterEngine can only be attached to one FlutterViewController at a time. Set FlutterEngine.viewController to nil before attaching it to another FlutterViewController. 

While on this topic, should we do this automatically in the FlutterViewController's dealloc too? Unsetting the engine's viewController property.

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: > While on this topic, should we do this automatically in the FlutterViewController's dealloc too? Unsetting the engine's viewController property.

That already happens, niling out an engine's view controller with their view controller is dealloc'd.

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: message updated.

**Comment by gaaclarke** [comment]
landing, previously only failed formatting, now it's passing.

**Pull request merged**